### PR TITLE
updated orion documentation with tourguide examples.

### PR DESCRIPTION
**Pull request by FiwareULPGC**
This PR updates the documentation related with Orion Context Broker. Now the documentation examples follow the model defined in the Tourguide Application.

A compiled version can be consulted at <http://testfiwaretour.readthedocs.org/en/refactor-orionv1-adapt_tourguide/development-context-aware-applications/introduction/>

### CODE REVIEW

File: doc/development-context-aware-applications/how-to-perform-geo-located-queries.md [hunk 1]
```diff
@@ -2,10 +2,10 @@ One very powerful feature in Context Broker GE is the ability to perform
 geo-located queries. You can query entities located inside (or outside)
 a region defined by a circle or a polygon.  
  For example, to query for all the restaurants within 13 km of the
-Madrid city center (identified by GPS coordinates 40.418889, -3.691944)
+Victoria Gasteiz city center (identified by GPS coordinates 42.846718, -2.671635)
 a Context Consumer application will use the following query:
 
-    POST <cb_host>:<cb_port>/v1/queryContext
+    POST <cb_host>:<cb_port>/v1/contextEntities
```
Comment (fgalan) [inline doc/development-context-aware-applications/how-to-perform-geo-located-queries.md]: Are you sure this is working? I'm getting:

```
{
  "orionError" : {
    "code" : "400",
    "reasonPhrase" : "Bad Request",
    "details" : "JSON Parse Error: unknown field: /entities"
  }
}
```

Note that with `/v1/queryContext` it works ok.

Comment (fgalan) [inline doc/development-context-aware-applications/how-to-perform-geo-located-queries.md]: I have used exactly this test script:

```
curl localhost:1026/v1/contextEntities -s -S --header 'Content-Type: application/json' --header 'Accept: application/json' -d @-  <<EOF
{
    "entities": [
        {
            "type": "Restaurant",
            "isPattern": "true",
            "id": ".*"
        }
    ],
    "restriction": {
        "scopes": [
            {
                "type": "FIWARE::Location",
                "value": {
                    "circle": {
                        "centerLatitude": 42.846718,
                        "centerLongitude": -2.671635,
                        "centerLatitude": "40.418889",
                        "centerLongitude": "-3.691944",
                        "radius": "13000"
                    }
                }
            }
        ]
    }
}
EOF
```

File: doc/development-context-aware-applications/how-to-perform-geo-located-queries.md [hunk 2]
```diff
@@ -20,12 +20,66 @@ a Context Consumer application will use the following query:
...
             ]
         }
     }
+
+
+
+To query for all restaurants inside a defined zone inside Victoria Gasteiz a Context Consumer application will use the following query: 
+
+
+    POST <cb_host>:<cb_port>/v1/contextEntities
```
Comment (fgalan) [inline doc/development-context-aware-applications/how-to-perform-geo-located-queries.md]: Same here.

File: doc/development-context-aware-applications/how-to-perform-geo-located-queries.md (hunk 1 above)
Comment (FiwareULPGC) [inline doc/development-context-aware-applications/how-to-perform-geo-located-queries.md]: You are right. We had an issue with the REST client that we was using. Thank you for the feedback!

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 3]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
-          "type": "string",
-          "value": "Calle Alcala 57, 28028 Madrid, Spain"
+          "type": "Restaurant",
+          "isPattern": "false",
+          "id": "Elizalde",
+          "attributes":
+          [
+            {
+              "name": "address",
+              "type": "",
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: strange that the type is the empty string. If so, probably it is a problem of the feeding Tour Guide App. Please, could you double check with Bitergia Team?

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 4]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
+                "type": "postalAddress",
+                "streetAddress": "Cuesta de las Cabras Aldapa 2",
+                "addressRegion": "Araba",
+                "addressLocality": "Alegría-Dulantzi",
+                "postalCode": "01240"
+              }
+            },
+            {
+              "name": "aggregateRating",
+              "type": "",
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: strange type empty string see above

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 5]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
+            },
+            {
+              "name": "department",
+              "type": "",
+              "value": "Franchise3"
+            },
+            {
+              "name": "description",
+              "type": "",
+              "value": "Restaurante de estilo sidrería ubicado en Alegria-Dulantzi. Además de su menú del día y carta, también ofrece menú de sidrería. El menú del día cuesta 9 euros. Los fines de semana la especialidad de la casa son las alubias con sacramentos. En lo que a bebidas se refiere, hay una amplia selección además de la sidra. Cabe destacar que se puede hacer txotx. La capacidad del establecimiento es de 50 personas pero la sidrería no dispone de aparcamiento.%5cn%5cnHORARIO: %5cn%5cnLunes a domingo: 9:00-17:00 y 19:00-23:00."
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: we can make some elipsis here for the sake of brevity and space

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 6]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
+            },
+            {
+              "name": "occupancyLevels",
+              "type": "PropertyValue",
+              "value": 0,
+              "metadatas":
+              [
+                {
+                  "name": "timestamp",
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: strange

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 7]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
+                  "type": "",
+                  "value": ""
+                },
+                {
+                  "name": "name",
+                  "type": "",
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: strange

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 8]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
-            "value": "4.2"
+      "contextElement" : {
+        "type" : "Restaurant",
+        "isPattern" : "false",
+        "id" : "Elizalde",
+        "attributes" : {
+          "address" : {
+            "type" : "",
+            "value" : {
+              "type" : "postalAddress",
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: are you using the latest version of the Tour Guide App, this should be PostalAddress

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 9]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
           },
-          "postal_address": {
-            "type": "string",
-            "value": "Calle Alcala 57, 28028 Madrid, Spain"
+          "capacity" : {
+            "type" : "PropertyValue",
+            "value" : 120,
+            "metadatas" : [
+              {
+                "name" : "name",
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: strange

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 10]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
-          "cousine_type": {
-            "type": "string",
-            "value": "french"
+          "department" : {
+            "type" : "",
+            "value" : "Franchise3"
+          },
+          "description" : {
+            "type" : "",
+            "value" : "Restaurante de estilo sidrería ubicado en Alegria-Dulantzi. Además de su menú del día y carta, también ofrece menú de sidrería. El menú del día cuesta 9 euros. Los fines de semana la especialidad de la casa son las alubias con sacramentos. En lo que a bebidas se refiere, hay una amplia selección además de la sidra. Cabe destacar que se puede hacer txotx. La capacidad del establecimiento es de 50 personas pero la sidrería no dispone de aparcamiento.%5cn%5cnHORARIO: %5cn%5cnLunes a domingo: 9:00-17:00 y 19:00-23:00."
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: elipsis

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 11]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
+          },
+          "occupancyLevels" : {
+            "type" : "PropertyValue",
+            "value" : 0,
+            "metadatas" : [
+              {
+                "name" : "timestamp",
+                "type" : "",
+                "value" : ""
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: strange

File: doc/development-context-aware-applications/introduction.md [hunk 12]
```diff
@@ -4,9 +4,7 @@ consume context information at large scale and exploit it to transform
 your application into a truly smart application.  
 
 Context information is represented through values assigned to attributes
-that characterize those entities relevant to your application. [The
-Context Broker GE (open source reference implementation:
-Orion)](http://catalogue.fiware.org/enablers/publishsubscribe-context-broker-orion-context-broker)
+that characterize those entities relevant to your application. [The Context Broker GE (open source reference implementation: Orion)](http://catalogue.fiware.org/enablers/publishsubscribe-context-broker-orion-context-broker)
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/introduction.md]: weird change, I cannot see the differences :(

File: doc/development-context-aware-applications/how-to-connect-to-external-context-providers.md [hunk 13]
```diff
@@ -24,26 +24,19 @@ advantages:
...
 consider an external restaurant booking system that is able to provide
-in real time the number of free places of a given restaurant and
-occupancy level (as a percentage). That system runs at
+in real time the occupancy level (as an absolute value). That system runs at
 http://booking.restaurants.foo.com and plays the role of a Context
 Provider.
 
-First, the Context Provider is registered as provider of free\_places
-and occupancy\_level for “LeBistro” restaurant:
+First, the Context Provider is registered as provider of occupancyLeves for Elizalde restaurant:
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-connect-to-external-context-providers.md]: nit: `occupancyLevels`

File: doc/development-context-aware-applications/how-to-connect-to-external-context-providers.md [hunk 14]
```diff
@@ -24,26 +24,19 @@ advantages:
...
-    }
-    POST <cb_host>:<cb_port>/v1/registry/contextEntities/type/Restaurant/id/LeBistro/attributes/occupancy_level
+    POST <cb_host>:<cb_port>/v1/registry/contextEntities/type/Restaurant/id/Elizalde/attributes/occupancyLevels
     {
       "duration" : "P1M",
       "providingApplication" : "http://booking.restaurants.foo.com"
     }
 
-Thus, each time free\_places or occupancy\_level is queried at Context
+Thus, each time occupancyLevels is queried at Context
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-connect-to-external-context-providers.md]: `occupancyLevels`

**Comment by jmcanterafonseca** [comment]
overall looks good but I'm a bit worried about data inconsistency, please check

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 15]
```diff
@@ -66,101 +81,201 @@ it in the following way:
...
+                  "type": "",
+                  "value": "occupancyLevels"
+                }
+              ]
+            },
+            {
+              "name": "position",
+              "type": "coords",
+              "value": "42.8404625, -2.5123277",
+              "metadatas":
```
Comment (fgalan) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: The usage of `location` to specify entity location is going to be deprecated by Orion 1.1.0 (released by the end of April), see https://github.com/telefonicaid/fiware-orion/pull/2071. Thus, I'd recommend to use the "new way" in the examples (based in using `geo:point` as attribute type).

File: doc/development-context-aware-applications/introduction.md (hunk 12 above)
Comment (FiwareULPGC) [inline doc/development-context-aware-applications/introduction.md]: We deleted new lines characters inside the link. We can undo this change if you want

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md (hunk 8 above)
Comment (FiwareULPGC) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: We are using the last docker image available at Docker Hub [link](https://hub.docker.com/r/fiware/tutorials.tourguide-app/builds/bdj2dbodbjmermzzxqbokj4/) . It seems to be up to date with the develop branch. We will rebuild and check it again

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 16]
```diff
@@ -4,163 +4,250 @@ exports, are said to play a **Context Producer** role.  As an example,
...
+          },
+          {
+              "name": "dateCreated",
+              "type": "date",
+              "value": "2016-04-20T18:19:33.00Z"
+          },
+          {
+              "name": "itemReviewed",
+              "type": "Restaurant",
+              "value": "Elizalde"
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: if we have changed the id, then here it cannot be Elizalde

**Comment by jmcanterafonseca** [comment]
please check the payload indentation and consistency. In v1 payloads everything is represented as a string so it is a surprise to see it as you represent it, are those the outputs provided by the real service?

**Comment by FiwareULPGC** [comment]
We have executed all the requests using curl again and we have copy-pasted the result from the console. We only have modified aggregateRating values because they are not updated correctly.

aggregateRating values are always:
```
 {
     "reviewCount" : 0,
     "ratingValue" : null
}
```

File: doc/development-context-aware-applications/how-to-update-and-query-context-information.md [hunk 17]
```diff
@@ -4,163 +4,251 @@ exports, are said to play a **Context Producer** role.  As an example,
...
-          "name" : "score",
-          "type" : "integer",
-          "value" : "4"
+      "attributes":
+      [
+        {
+          "name": "author",
+          "type": "Person",
+          "value": "“Client1234”"
+        },
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/how-to-update-and-query-context-information.md]: nit: "Client1234"

**Comment by jmcanterafonseca** [comment]
LGTM. @FiwareULPGC  Please fix the nit, double quotes on Client id and then merge yourself squashing,

**Pull request merged**